### ISSUE TITLE
How to specify discard do binds hint?

### ISSUE THREAD

**Issue by chshersh** [ndmitchell/hlint#779]
Hi! I would like to specify the following rule with `hlint`:

```yaml
- warn:
    lhs: _ <- foo
    rhs: foo_
```

However, when I try to run `hlint` (version 2.2.2), I see the following error:

```
hlint: user error (Failed to read YAML configuration file ~/haskell/sandbox/.hlint.yaml
  Aeson exception:
Error in $: Error when decoding YAML, Failed to parse Parse error: <-, when parsing:
  _ <- foo
Along path: root 0 warn lhs
When at: String
- warn:
    lhs: _ <- foo
    rhs: foo_
)

```

Is there any way to specify discarding of monadic computation inside `do`-block?

**Comment by ndmitchell** [comment]
No, there isn't at the moment. For certain names it is built in (see https://github.com/ndmitchell/hlint/blob/master/src/Hint/Monad.hs#L68), but no way of extending it. Is your specific `foo` common?

**Comment by chshersh** [comment]
@ndmitchell No, this is my own function in a library, not common. In that case, I think this would be a good feature of `hlint`.

**Comment by ndmitchell** [comment]
To have it as a parameterisable feature you would need to extend the `.hlint.yaml` file to specify the set of functions and then feed it into the hint.
